python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -109,6 +109,11 @@ def test_all_none_category(self):
         df['a'] = df['a'].astype('category')
         self._check_pandas_roundtrip(df)
 
+    def test_non_string_columns(self):
```

Comment:
I suppose we can leave this as the only test right now and say that anything other integers or strings is undefined behavior.